### PR TITLE
[#2769][WebUI] Simplified torrent file upload UX

### PR DESCRIPTION
**Pull request by bendykst**
Previously, the process for uploading a file in the Web-UI required three
steps. Click 'File' to open the 'Add from File' window.  Click 'Browse' to
select the file. Finally, click 'Add' to upload the file. These steps have
been combined into one, making the process much easier. Now, clicking 'File'
opens the file browser directly. After a file is selected, it is uploaded
automatically.

### CODE REVIEW

File: deluge/ui/web/js/deluge-all/add/AddWindow.js [hunk 1]
```diff
@@ -52,7 +79,7 @@ Deluge.add.AddWindow = Ext.extend(Deluge.add.Window, {
         this.addButton(_('Add'), this.onAddClick, this);
 
         function torrentRenderer(value, p, r) {
-            if (r.data['info_hash']) {
+            if (r.data.info_hash) {
```
Comment (cas--) [inline deluge/ui/web/js/deluge-all/add/AddWindow.js]: Can you use original syntax, this is data not an attribute.

Comment (bendykst) [inline deluge/ui/web/js/deluge-all/add/AddWindow.js]: Sure, will do.

File: deluge/ui/web/js/deluge-all/add/AddWindow.js [hunk 2]
```diff
@@ -215,8 +287,8 @@ Deluge.add.AddWindow = Ext.extend(Deluge.add.Window, {
             });
             this.list.getStore().remove(r);
         } else {
-            r.set('info_hash', info['info_hash']);
-            r.set('text', info['name']);
+            r.set('info_hash', info.info_hash);
```
Comment (cas--) [inline deluge/ui/web/js/deluge-all/add/AddWindow.js]: Same as above, revert to original syntax

File: deluge/ui/web/js/deluge-all/add/AddWindow.js [hunk 3]
```diff
@@ -117,12 +159,16 @@ Deluge.add.AddWindow = Ext.extend(Deluge.add.Window, {
...
         });
 
+        this.fileUploadForm = Ext.getCmp('fileUploadForm').getForm();
         this.optionsPanel = this.add(new Deluge.add.OptionsPanel());
         this.on('hide', this.onHide, this);
         this.on('show', this.onShow, this);
     },
 
     clear: function() {
+        this.fileUploadForm.reset(); // Reset form so that onFileSelected handler
```
Comment (cas--) [inline deluge/ui/web/js/deluge-all/add/AddWindow.js]: Iirc multi-line comment will mess with the gettext parsing

**Comment by cas--** [comment]
Thanks for this, merged: [005db434f80c93423]

**Comment by th0ma7** [comment]
@bendykst could this have broken the ability to upload multiple torrent in one shot?  Previously I used to be able to select multiple torrent and they would load.  Now they do appear as selected but they never load into the main interface...

**Comment by cas--** [comment]
@th0ma7 See: https://github.com/deluge-torrent/deluge/pull/481#issuecomment-3629072437

**Pull request closed without merging**